Ball of Duty Server/Domain/Entities/Character.cs

```diff
@@ -45,18 +47,34 @@ public void AddKill(Character victim)
             {
                 HighScore = Score;
             }
+            UpdateServerStats();
             NotifyObservers(Observation.ACQUISITION_OF_GOLD, victim);
         }
 
+        public void DecayAndRegen()
```

Comment:
LightEvent is VERY cheap. you really should keep logic like this separate. make one event for each, that will also futureproof the design in the case that decay and regen shouldnt happen at the same interval